src/dotnet/CarbonAware.WebApi/Controllers/CarbonAwareController.cs

```diff
@@ -24,7 +24,11 @@ public CarbonAwareController(ILogger<CarbonAwareController> logger, ICarbonAware
     [HttpGet("bylocations/best")]
     public async Task<IActionResult> GetBestEmissionsDataForLocationsByTime([FromQuery(Name = "locations")] string[] locations, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
     {
-        IEnumerable<Location> locationEnumerable = locations.Select(loc => new Location(){ RegionName = loc });
+        //The LocationType and CloudProvider are hardcoded for now. Ideally this should be received from the request or configuration 
+        IEnumerable<Location> locationEnumerable = locations.Select(loc => new Location()
+                                                                            { RegionName = loc, 
+                                                                            LocationType=LocationType.CloudProvider, 
```

Comment:
I think this case is worth handling as part of this PR.  This endpoint should probably not hardcode `Azure`.

Perhaps we could hardcode `LocationType` as you have it, leave `CloudProvider` null and update the logic in the `AzureLocationSource` to not throw an exception if no provider is specified.  Thoughts?